File: eproject-java-master/src/main/java/com/example/demo/entities/Staff.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.example.demo.entities;

import java.io.Serializable;
import java.util.Date;
import javax.persistence.Basic;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQueries;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.validation.constraints.Size;
import javax.xml.bind.annotation.XmlRootElement;
import org.hibernate.annotations.*;

/**
 *
 * @author Red
 */
@Entity
@Table(name = "staff")
@XmlRootElement
@NamedQueries({
    @NamedQuery(name = "Staff.findAll", query = "SELECT s FROM Staff s"),
    @NamedQuery(name = "Staff.findById", query = "SELECT s FROM Staff s WHERE s.id = :id"),
    @NamedQuery(name = "Staff.findByCode", query = "SELECT s FROM Staff s WHERE s.code = :code"),
    @NamedQuery(name = "Staff.findByFirstname", query = "SELECT s FROM Staff s WHERE s.firstname = :firstname"),
    @NamedQuery(name = "Staff.findByLastname", query = "SELECT s FROM Staff s WHERE s.lastname = :lastname"),
    @NamedQuery(name = "Staff.findByDepartment", query = "SELECT s FROM Staff s WHERE s.department = :department"),
    @NamedQuery(name = "Staff.findByIsManager", query = "SELECT s FROM Staff s WHERE s.isManager = :isManager"),
    @NamedQuery(name = "Staff.findByJoinedAt", query = "SELECT s FROM Staff s WHERE s.joinedAt = :joinedAt"),
    @NamedQuery(name = "Staff.findByOffDate", query = "SELECT s FROM Staff s WHERE s.offDate = :offDate"),
    @NamedQuery(name = "Staff.findByDob", query = "SELECT s FROM Staff s WHERE s.dob = :dob"),
    @NamedQuery(name = "Staff.findByGender", query = "SELECT s FROM Staff s WHERE s.gender = :gender"),
    @NamedQuery(name = "Staff.findByRegional", query = "SELECT s FROM Staff s WHERE s.regional = :regional"),
    @NamedQuery(name = "Staff.findByPhoneNumber", query = "SELECT s FROM Staff s WHERE s.phoneNumber = :phoneNumber"),
    @NamedQuery(name = "Staff.findByEmail", query = "SELECT s FROM Staff s WHERE s.email = :email"),
    @NamedQuery(name = "Staff.findByPassword", query = "SELECT s FROM Staff s WHERE s.password = :password"),
    @NamedQuery(name = "Staff.findByIdNumber", query = "SELECT s FROM Staff s WHERE s.idNumber = :idNumber"),
    @NamedQuery(name = "Staff.findByPhoto", query = "SELECT s FROM Staff s WHERE s.photo = :photo"),
    @NamedQuery(name = "Staff.findByIdPhoto", query = "SELECT s FROM Staff s WHERE s.idPhoto = :idPhoto"),
    @NamedQuery(name = "Staff.findByIdPhotoBack", query = "SELECT s FROM Staff s WHERE s.idPhotoBack = :idPhotoBack"),
    @NamedQuery(name = "Staff.findByDayOfLeave", query = "SELECT s FROM Staff s WHERE s.dayOfLeave = :dayOfLeave"),
    @NamedQuery(name = "Staff.findByNote", query = "SELECT s FROM Staff s WHERE s.note = :note"),
    @NamedQuery(name = "Staff.findByCreatedAt", query = "SELECT s FROM Staff s WHERE s.createdAt = :createdAt"),
    @NamedQuery(name = "Staff.findByCreatedBy", query = "SELECT s FROM Staff s WHERE s.createdBy = :createdBy"),
    @NamedQuery(name = "Staff.findByUpdatedAt", query = "SELECT s FROM Staff s WHERE s.updatedAt = :updatedAt"),
    @NamedQuery(name = "Staff.findByUpdatedBy", query = "SELECT s FROM Staff s WHERE s.updatedBy = :updatedBy"),
    @NamedQuery(name = "Staff.findByStatus", query = "SELECT s FROM Staff s WHERE s.status = :status")})
public class Staff implements Serializable {

    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Basic(optional = false)
    @Column(name = "id")
    private Integer id;
    @Size(max = 30)
    @Column(name = "code")
    private String code;
    @Size(max = 255)
    @Column(name = "firstname")
    private String firstname;
    @Size(max = 255)
    @Column(name = "lastname")
    private String lastname;
    @Column(name = "department")
    private Integer department;
    @Column(name = "is_manager")
    private Boolean isManager;
    @Column(name = "joined_at")
    @Temporal(TemporalType.DATE)
    private Date joinedAt;
    @Column(name = "off_date")
    @Temporal(TemporalType.DATE)
    private Date offDate;
    @Column(name = "dob")
    @Temporal(TemporalType.DATE)
    private Date dob;
    @Column(name = "gender")
    private Short gender;
    @Column(name = "regional")
    private Integer regional;
    @Size(max = 50)
    @Column(name = "phone_number")
    private String phoneNumber;
    // @Pattern(regexp="[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?", message="Invalid email")//if the field contains email address consider using this annotation to enforce field validation
    @Size(max = 50)
    @Column(name = "email")
    private String email;
    @Size(max = 255)
    @Column(name = "password")
    private String password;
    @Size(max = 15)
    @Column(name = "id_number")
    private String idNumber;
    @Column(name = "identityIssueDate")
    @Temporal(TemporalType.DATE)
    private Date identity_issue_date;
    @Size(max = 255)
    @Column(name = "photo")
    private String photo;
    @Size(max = 255)
    @Column(name = "id_photo")
    private String idPhoto;
    @Size(max = 255)
    @Column(name = "id_photo_back")
    private String idPhotoBack;
    // @Max(value=?)  @Min(value=?)//if you know range of your decimal fields consider using these annotations to enforce field validation
    @Column(name = "day_of_leave")
    private Float dayOfLeave;
    @Size(max = 1000)
    @Column(name = "note")
    private String note;
    @Column(name = "created_at")
    @Temporal(TemporalType.TIMESTAMP)
    @CreationTimestamp
    private Date createdAt;
    @Column(name = "created_by")
    private Integer createdBy;
    @Column(name = "updated_at")
    @Temporal(TemporalType.TIMESTAMP)
    @UpdateTimestamp
    private Date updatedAt;
    @Column(name = "updated_by")
    private Integer updatedBy;
    @Column(name = "status")
    private boolean status;

    public Staff() {
    }

    public Staff(Integer id) {
        this.id = id;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public Integer getDepartment() {
        return department;
    }

    public void setDepartment(Integer department) {
        this.department = department;
    }

    public Boolean getIsManager() {
        return isManager;
    }

    public void setIsManager(Boolean isManager) {
        this.isManager = isManager;
    }

    public Date getJoinedAt() {
        return joinedAt;
    }

    public void setJoinedAt(Date joinedAt) {
        this.joinedAt = joinedAt;
    }

    public Date getOffDate() {
        return offDate;
    }

    public void setOffDate(Date offDate) {
        this.offDate = offDate;
    }

    public Date getDob() {
        return dob;
    }

    public void setDob(Date dob) {
        this.dob = dob;
    }

    public Short getGender() {
        return gender;
    }

    public void setGender(Short gender) {
        this.gender = gender;
    }

    public Integer getRegional() {
        return regional;
    }

    public void setRegional(Integer regional) {
        this.regional = regional;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getIdNumber() {
        return idNumber;
    }

    public void setIdNumber(String idNumber) {
        this.idNumber = idNumber;
    }

    public void setIdentity_issue_date(Date identity_issue_date) {
        this.identity_issue_date = identity_issue_date;
    }

    public Date getIdentity_issue_date() {
        return identity_issue_date;
    }
    

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }

    public String getIdPhoto() {
        return idPhoto;
    }

    public void setIdPhoto(String idPhoto) {
        this.idPhoto = idPhoto;
    }

    public String getIdPhotoBack() {
        return idPhotoBack;
    }

    public void setIdPhotoBack(String idPhotoBack) {
        this.idPhotoBack = idPhotoBack;
    }

    public Float getDayOfLeave() {
        return dayOfLeave;
    }

    public void setDayOfLeave(Float dayOfLeave) {
        this.dayOfLeave = dayOfLeave;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Integer getCreatedBy() {
        return createdBy;
    }

    public void setCreatedBy(Integer createdBy) {
        this.createdBy = createdBy;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    public Integer getUpdatedBy() {
        return updatedBy;
    }

    public void setUpdatedBy(Integer updatedBy) {
        this.updatedBy = updatedBy;
    }
    
    public void setStatus(boolean status) {
        this.status = status;
    }

    public boolean getStatus() {
        return status;
    }


    @Override
    public int hashCode() {
        int hash = 0;
        hash += (id != null ? id.hashCode() : 0);
        return hash;
    }

    @Override
    public boolean equals(Object object) {
        // TODO: Warning - this method won't work in the case the id fields are not set
        if (!(object instanceof Staff)) {
            return false;
        }
        Staff other = (Staff) object;
        if ((this.id == null && other.id != null) || (this.id != null && !this.id.equals(other.id))) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "com.example.demo.entities.Staff[ id=" + id + " ]";
    }
    
}

File: futurehrm_android_app/lib/models/staff.dart
import 'dart:core';

import 'package:hive/hive.dart';

part 'staff.g.dart';

@HiveType(typeId: 1)
class Staff {
  @HiveField(0)
  int? id;
  @HiveField(1)
  String? code;
  @HiveField(2)
  String? firstname;
  @HiveField(3)
  String? lastname;
  @HiveField(4)
  int? department;
  @HiveField(5)
  bool? isManager;
  @HiveField(6)
  DateTime? joinedAt;
  @HiveField(7)
  DateTime? offDate;
  @HiveField(8)
  DateTime? dob;
  @HiveField(9)
  int? gender;
  @HiveField(10)
  int? regional;
  @HiveField(11)
  String? phoneNumber;
  @HiveField(12)
  String? email;
  @HiveField(13)
  String? password;
  @HiveField(14)
  String? idNumber;
  @HiveField(15)
  DateTime? identity_issue_date;
  @HiveField(16)
  String? photo;
  @HiveField(17)
  String? idPhoto;
  @HiveField(18)
  String? idPhotoBack;
  @HiveField(19)
  double? dayOfLeave;
  @HiveField(20)
  String? note;
  @HiveField(21)
  DateTime? createdAt;
  @HiveField(22)
  int? createdBy;
  @HiveField(23)
  DateTime? updatedAt;
  @HiveField(24)
  int? updatedBy;
  @HiveField(25)
  bool? status;

  Staff({
    this.id,
    this.code,
    this.firstname,
    this.lastname,
    this.department,
    this.isManager,
    this.joinedAt,
    this.offDate,
    this.dob,
    this.gender,
    this.regional,
    this.phoneNumber,
    this.email,
    this.password,
    this.idNumber,
    this.identity_issue_date,
    this.photo,
    this.idPhoto,
    this.idPhotoBack,
    this.dayOfLeave,
    this.note,
    this.createdAt,
    this.createdBy,
    this.updatedAt,
    this.updatedBy,
    this.status,
  });

  factory Staff.fromMap(Map<String, dynamic> json) {
    return Staff(
      id: json['id'],
      code: json['code'],
      firstname: json['firstname'],
      lastname: json['lastname'],
      department: json['department'],
      isManager: json['isManager'],
      joinedAt: json['joinedAt'] != null
          ? DateTime.fromMillisecondsSinceEpoch(json['joinedAt'])
          : null,
      offDate: json['offDate'] != null
          ? DateTime.fromMillisecondsSinceEpoch(json['offDate'])
          : null,
      dob: json['dob'] != null
          ? DateTime.fromMillisecondsSinceEpoch(json['dob'])
          : null,
      gender: json['gender'],
      regional: json['regional'],
      phoneNumber: json['phoneNumber'],
      email: json['email'],
      password: json['password'],
      idNumber: json['idNumber'],
      identity_issue_date: json['identity_issue_date'] != null
          ? DateTime.fromMillisecondsSinceEpoch(json['identity_issue_date'])
          : null,
      photo: json['photo'],
      idPhoto: json['idPhoto'],
      idPhotoBack: json['idPhotoBack'],
      dayOfLeave: json['dayOfLeave'],
      note: json['note'],
      createdAt: json['createdAt'] != null
          ? DateTime.fromMillisecondsSinceEpoch(json['createdAt'])
          : null,
      createdBy: json['createdBy'],
      updatedAt: json['updatedAt'] != null
          ? DateTime.fromMillisecondsSinceEpoch(json['updatedAt'])
          : null,
      updatedBy: json['updatedBy'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      "id": id,
      "code": code,
      "firstname": firstname,
      "lastname": lastname,
      "department": department,
      "isManager": isManager,
      "joinedAt": joinedAt?.toIso8601String(),
      "offDate": offDate?.toIso8601String(),
      "dob": dob?.toIso8601String(),
      "gender": gender,
      "regional": regional,
      "phoneNumber": phoneNumber,
      "email": email,
      "password": password,
      "idNumber": idNumber,
      "identity_issue_date": identity_issue_date?.toIso8601String(),
      "photo": photo,
      "idPhoto": idPhoto,
      "idPhotoBack": idPhotoBack,
      "dayOfLeave": dayOfLeave,
      "note": note,
      "createdAt": createdAt?.toIso8601String(),
      "createdBy": createdBy,
      "updatedAt": updatedAt?.toIso8601String(),
      "updatedBy": updatedBy,
      "status": status,
    };
  }
}
